src/coreclr/jit/importercalls.cpp

```diff
@@ -6111,8 +6202,41 @@ void Compiler::impMarkInlineCandidate(GenTree*               callNode,
 {
     GenTreeCall* call = callNode->AsCall();
 
-    // Do the actual evaluation
-    impMarkInlineCandidateHelper(call, exactContextHnd, exactContextNeedsRuntimeLookup, callInfo, ilOffset);
+    // Call might not have an inline candidate info yet (will be set by impMarkInlineCandidateHelper)
+    // so we assume there is always a least one candidate:
+    //
+
+    if (call->IsGuardedDevirtualizationCandidate())
+    {
+        const uint8_t candidatesCount = call->GetInlineCandidatesCount();
+        assert(candidatesCount > 0);
+        for (uint8_t candidateId = 0; candidateId < candidatesCount; candidateId++)
+        {
+            // Do the actual evaluation
+            bool success = impMarkInlineCandidateHelper(call, candidateId, exactContextHnd,
```

Comment:
Maybe hoist the `InlineResult` creation here and pass it into `impMarkInlineCandidateHelper`, then you can just check that result for failure and not modify `impMarkInlineCandidateHelper` to return a bool?

Reply:
Good idea, addressed!